src/components/select/filterable-select/filterable-select.spec.js

```diff
@@ -1004,6 +1033,40 @@ describe("FilterableSelect", () => {
           expect(onOpenFn).toHaveBeenCalled();
         });
 
+        it("then it should not be invoked when the component rerenders", () => {
+          const Component = () => {
+            const [callCounter, setCallCounter] = useState(0);
+
+            return (
+              <>
+                <div id="call-counter">{callCounter}</div>
+                <FilterableSelect
+                  name="testSelect"
+                  id="testSelect"
+                  openOnFocus
+                  onOpen={() => {
+                    if (callCounter < 2) {
+                      setCallCounter(callCounter + 1);
+                    }
+                  }}
```

Comment:
Without this condition the component goes into an infinite rerender loop

Reply:
Oops I overlooked that the component will re-render when the div text changes 🤦🏼 Though I still don't think that this test is verifying the `onOpen` callback hasn't been invoked again because the changed incremented counter text will trigger a re-render anyway.

Reply:
I forgot to mention that it actually ended up in an infinite loop before my fix. So I've left this guard to ensure that if someone in the future breaks it then they get an information about which unit test fails instead of a vague `Maximum call stack size exceeded...` message

Reply:
Ah thanks for clarifying this @mkrds 👍🏼 If this test no longer goes into an infinite loop after your fix, what do you think about adding a comment instead of the condition, or changing the condition to `if (callCounter < 3)`? 

That way we can safely assume if the rendered counter text is 2, then `onOpen` has been incorrectly called again after the re-render

Reply:
Realised the text can become 2 and isn't blocked by the condition because its a less-than condition 🤦🏼 This makes my comment irrelevant so happy to accept 👍🏼